rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSource.java

```diff
@@ -189,6 +235,10 @@ public void setVirtual(boolean virtual) {
          * 		 since several NS can deploy on the same node
          */
         public String getId() {
+            return generateId(this.sourceName, this.hostName);
+        }
+
+        public static String generateId(String sourceName, String hostName) {
             return sourceName + "-host-" + hostName;
         }
 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 17 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)